ethstore/src/accounts_dir/disk.rs

```diff
@@ -153,22 +153,31 @@ impl<T> DiskDirectory<T> where T: KeyFileManager {
 		)
 	}
 
-	/// insert account with given file name
-	pub fn insert_with_filename(&self, account: SafeAccount, filename: String) -> Result<SafeAccount, Error> {
+
+	/// insert account with given filename. if the filename is a duplicate of any stored account and dedup is set to
+	/// true, a random suffix is appended to the filename.
+	pub fn insert_with_filename(&self, account: SafeAccount, mut filename: String, dedup: bool) -> Result<SafeAccount, Error> {
```

Comment:
why is `dedup` optional and not a default behaviour?

Reply:
Afair this function is also used to update the key file (for instance after a name change or meta change).